rpm/generic/zfs.spec.in

```diff
@@ -88,12 +88,15 @@ BuildRequires:  libudev-devel
 BuildRequires:  libattr-devel
 BuildRequires:  openssl-devel
 Requires:       openssl
+%if 0%{?_systemd}
+BuildRequires: systemd
+%endif
```

Comment:
I had to move this inside the first `%if 0%{?rhel}%{?fedora}%{?suse_version}` condition, the `rpm` binary on my Debian9 doesn't know how to handle the "BuildRequires" tag (probably because we're missing the rpmdb on Debian-based distros).

This is from `gdb` attached to the `rpmbuild` process:

```
Breakpoint 1, unsatisfiedDepend (ts=0x555555797b10, dcache=0x5555557eb770, dep=0x5555557e9a10) at depends.c:784
warning: Source file is more recent than executable.
784		rc = 1;
(gdb) bt
#0  unsatisfiedDepend (ts=0x555555797b10, dcache=0x5555557eb770, dep=0x5555557e9a10) at depends.c:784
#1  0x00007ffff7b8823f in checkDS (ts=0x555555797b10, dcache=0x5555557eb770, te=0x5555557d0120, pkgNEVRA=0x5555557cb280 "zfs-0.7.0-213_gf6940bb9e.x86_64", ds=0x5555557e9a10, tscolor=3) at depends.c:808
#2  0x00007ffff7b8891f in rpmtsCheck (ts=0x555555797b10) at depends.c:936
#3  0x00007ffff771b4c7 in rpmSpecCheckDeps (ts=0x555555797b10, spec=0x5555557982a0) at spec.c:322
#4  0x0000555555556b7f in checkSpec (ts=0x555555797b10, spec=0x5555557982a0) at rpmbuild.c:234
#5  0x000055555555758d in buildForTarget (ts=0x555555797b10, arg=0x55555578ac50 "SPECS/zfs.spec", ba=0x55555575a9e0 <rpmBTArgs>) at rpmbuild.c:453
#6  0x00005555555576ed in build (ts=0x555555797b10, arg=0x55555578ac50 "SPECS/zfs.spec", ba=0x55555575a9e0 <rpmBTArgs>, rcfile=0x0) at rpmbuild.c:490
#7  0x0000555555557cc2 in main (argc=18, argv=0x7fffffffea98) at rpmbuild.c:626
(gdb) p *dep
$1 = {pool = 0x5555557d0240, Type = 0x7ffff7bc1e61 "Requires", DNEVR = 0x555555826940 "R systemd", N = 0x5555557e9a70, EVR = 0x5555557e9ab0, Flags = 0x5555557e9ad0, Color = 0x0, tagN = 1049, Count = 1, 
  instance = 0, i = 0, nopromote = 1, nrefs = 1, ti = 0x0}
(gdb) fin
Run till exit from #0  unsatisfiedDepend (ts=0x555555797b10, dcache=0x5555557eb770, dep=0x5555557e9a10) at depends.c:784
0x00007ffff7b8823f in checkDS (ts=0x555555797b10, dcache=0x5555557eb770, te=0x5555557d0120, pkgNEVRA=0x5555557cb280 "zfs-0.7.0-213_gf6940bb9e.x86_64", ds=0x5555557e9a10, tscolor=3) at depends.c:808
808		if (unsatisfiedDepend(ts, dcache, ds) == is_problem)
Value returned is $2 = 1
(gdb) 
```